examples/swdump.rs

```diff
@@ -296,6 +296,10 @@ fn run(opts: Opts) -> Result<(), SwdError> {
     let khz = cmp::min(opts.speed.unwrap_or(200), 0xfffe);
     probe.set_speed(CommunicationSpeed::khz(khz).unwrap())?;
 
+    if probe.capabilities().contains(Capabilities::REGISTER) {
+        probe.register(Connection::default())?;
+    }
```

Comment:
Can this be done implicitly when the probe is opened, or are there any ordering requirements?

Reply:
If so, it might be good to make this purely an implementation detail, without exposing an API for it

Reply:
I don't think there are, but I did it this way in case someone eventually wanted to fully use this functionality, with real unique IDs and stuff (after adding more methods)

Reply:
Hmm, that's right. I think APIs like `open_usb` and `open_by_serial` should still do it automatically. Then later we can add more advanced functions once we figure out a more complete API for registration.